testutil/base.go

```diff
@@ -60,3 +62,31 @@ func (s *BaseTest) TearDownTest(c *check.C) {
 func (s *BaseTest) AddCleanup(f func()) {
 	s.cleanupHandlers = append(s.cleanupHandlers, f)
 }
+
+// Mocks any mutable object with a compliant substitute, and insures it will be restored after the test
+func (s *BaseTest) MockAny(mockedByPtr interface{}, mock interface{}) {
+	restore := MockAny(mockedByPtr, mock)
+	s.AddCleanup(restore)
+}
+
+// Mocks any mutable object with a compliant substitute, and returns a restore function
+func MockAny(mockedByPtr interface{}, mock interface{}) (restore func()) {
+	mockedPtr, mockValue := reflect.ValueOf(mockedByPtr), reflect.ValueOf(mock)
+
+	if mockedPtr.Type().Kind() != reflect.Ptr {
+		panic("Mock: first argument shall be a pointer to the mocked element!")
+	}
+
+	if mockedPtr.Elem().Type() != mockValue.Type() {
```

Comment:
Since you are using this a few times, I'd declare a it as a avariable:
```
mockedValue := mockedPtr.Elem()
```

Reply:
Totally agree with, this PR is a bit of draft. The idea was to start a conversation. That is why the names are not the best. I also kept few comments for you to review and compare the differences.

And you are absolutely it shall be called `Mock()`.

Reply:
Sure, great point. :pray: